src/detect-snmp-community.c

```diff
@@ -0,0 +1,218 @@
+/* Copyright (C) 2015-2017 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+/**
+ * \file
+ *
+ * \author Pierre Chifflier <chifflier@wzdftpd.net>
+ *
+ * Set up of the "snmp_community" keyword to allow content
+ * inspections on the decoded snmp community.
+ */
+
+#include "suricata-common.h"
+#include "conf.h"
+#include "detect.h"
+#include "detect-parse.h"
+#include "detect-engine.h"
+#include "detect-engine-content-inspection.h"
+#include "detect-snmp-community.h"
+#include "app-layer-parser.h"
+
+#include "rust-snmp-snmp-gen.h"
+#include "rust-snmp-detect-gen.h"
+
+static int DetectSNMPCommunitySetup(DetectEngineCtx *, Signature *,
+    const char *);
+static int DetectEngineInspectSNMPCommunity(ThreadVars *tv,
+    DetectEngineCtx *de_ctx, DetectEngineThreadCtx *det_ctx,
+    const Signature *s, const SigMatchData *smd,
+    Flow *f, uint8_t flags, void *alstate, void *txv, uint64_t tx_id);
+static void DetectSNMPCommunityRegisterTests(void);
+static int g_snmp_rust_id = 0;
+
+void DetectSNMPCommunityRegister(void)
+{
+    sigmatch_table[DETECT_AL_SNMP_COMMUNITY].name = "snmp.community";
+    sigmatch_table[DETECT_AL_SNMP_COMMUNITY].alias = "snmp_community";
```

Comment:
this alias can be removed. It's used to provide backward compat in other cases, but here it's not need as the code and functionality is all new